### PR TITLE
feat: XYK integration tests 

### PR DESCRIPTION
**Pull request by martinfridrich**
Added integration tests to check if pool account is added/removed from duster whitelist

### CODE REVIEW

File: integration-tests/src/xyk.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+use basilisk_runtime::{DustRemovalWhitelist, Origin, XYK};
+use hydradx_traits::AMM;
+use primitives::{asset::AssetPair, Price};
+use xcm_emulator::TestExt;
+
+use frame_support::{assert_ok, traits::Contains};
+
+#[test]
+fn pair_account_should_be_added_into_whitelist_when_pool_is_created() {
```
Comment (dmoka) [inline integration-tests/src/xyk.rs:13]: great job with the test names, descriptive ones! 👌

Comment (dmoka) [inline integration-tests/src/xyk.rs:13]: great job with the test names, descriptive ones! 👌

File: runtime/basilisk/src/lib.rs [hunk 2]
```diff
@@ -138,7 +138,6 @@ impl Contains<Call> for BaseFilter {
 
 		#[allow(clippy::match_like_matches_macro)]
 		match call {
-			Call::XYK(_) => false,
```
Comment (dmoka) [inline runtime/basilisk/src/lib.rs:141]: Just for my understanding: This is basically the filter we were talking about, where we can disable features for our runtime? So in this case, we enable XYK as this line is removed...

**Review by enthusiastmartin — APPROVED** [review]

File: runtime/basilisk/src/lib.rs (hunk 2 above)
Comment (martinfridrich) [reply runtime/basilisk/src/lib.rs:141]: yes

**Pull request merged**